crates/conductor_lib/src/conductor/base.rs

```diff
@@ -1538,29 +1539,44 @@ fn run_interface(
     kill_switch: Receiver<()>,
 ) -> Result<(Broadcaster, thread::JoinHandle<()>), String> {
     use crate::interface_impls::{http::HttpInterface, websocket::WebsocketInterface};
-    match interface_config.driver {
-        InterfaceDriver::Websocket { port } => with_port_heuristic(
-            port,
-            interface_config.choose_free_port.unwrap_or(false),
-            || {
-                let r = WebsocketInterface::new(port).run(handler, kill_switch);
-                println!("{}", magic_port_binding_string(&interface_config.id, port));
-                r
-            },
-        ),
-        InterfaceDriver::Http { port } => with_port_heuristic(
-            port,
-            interface_config.choose_free_port.unwrap_or(false),
-            || {
-                let r = HttpInterface::new(port).run(handler, kill_switch);
-                println!("{}", magic_port_binding_string(&interface_config.id, port));
-                r
-            },
-        ),
 
+    match interface_config.driver {
+        InterfaceDriver::Websocket { port } => {
+            let port = if interface_config.choose_free_port.unwrap_or(false) {
```

Comment:
I was wondering if choose_free_port needs to be an option? I'm thinking it is either you choose the free port or you don't right?

Reply:
Its nicer if its an option right because then Serde can still deserialize it if the field isn't there?

Reply:
We could use #[serde(default)] that way it default to false already

Reply:
oh yeah nice